src/prefect/engine/results/azure_result.py

```diff
@@ -52,8 +52,19 @@ def initialize_service(self) -> None:
         if not connection_string and self.connection_string_secret:
             connection_string = Secret(self.connection_string_secret).get()
 
+        if connection_string is None or any(
```

Comment:
I added this after mypy test failed complaining that connection_string is optional and that the list comprehension would fail otherwise. I am not sure why mypy did not complan (or remind me) in the respective other function. I aligned both functions now by having the None check in both of them. If we would not have the None check the code could crash if None is given

Reply:
I think if you want to perform a None check then I think using `and` makes more sense:
```python
if connection_string is not None and any(x in connection_string for x in ["AccountKey=", "SharedAccessSignature="]):
    ...
```
This way you'll still attempt to perform default authentication if a connection string is not provided.

Reply:
Actually, what I wanted to achieve here is that
1) if no connection string is given, do not run DefaultAzureCredential, since without connection string I do not need any authentication as I do not have any valid storage defined, **OR**
2) if connection string is given and connection string already includes a key, do not run DefaultAzureCredential as I have already a valid authentication.

Reply:
Your suggestion is of course correct from a "None"-guard perspective and if you prefer that solution I probably could change it also.

Technically, it would mean that I would run an authentication for a connection string that is None which makes not much sense.
Practically, a None connection string would not reach my code anyway as we would run into an Exceptions beforehand.

Reply:
Thanks for that explanation! I understand the interaction between the connection string and default credentials better now. Since a connection string is necessary in all cases, I think it would make sense to raise and exception if `connection_string` is `None`. Then you'd be able to remove the None check before inspecting the connection string. We'd also be able to provide a more useful error message detailing where the connection string needs to be set.

Reply:
I removed the None check and an exception added. Also added respective testcase.